### PR TITLE
Admin defines import/export for electricity & heat

### PR DESCRIPTION
**Pull request by dleard**
- Add radio widges for each import/export
- Fix mutations & functions
- Fix search function

### CODE REVIEW

File: app/containers/Products/ProductCreatorContainer.tsx [hunk 1]
```diff
@@ -27,11 +27,14 @@ export const ProductCreator: React.FunctionComponent<Props> = ({
           units: e.formData.units,
           state: 'active',
           parent: [null],
-          requiresEmissionAllocation: e.formData.requiresEmissionAllocation
+          requiresEmissionAllocation: e.formData.requiresEmissionAllocation,
+          includesImportedElectricity: e.formData.includesImportedElectricity,
+          includesExportedElectricity: e.formData.includesExportedElectricity,
+          includesImportedHeat: e.formData.includesImportedHeat,
+          includesExportedHeat: e.formData.includesExportedHeat
```
Comment (matthieu-foucault) [inline app/containers/Products/ProductCreatorContainer.tsx]: Maybe we should rename those for more clarity (sorry, I should have thought of that before): 
includesImportedElectricity -> addImportedElectricityEmissions
includesExportedElectricity -> substractExportedElectricityEmissions
includesImportedHeat -> addImportedHeatEmissions
includesExportedHeat -> substractExportedHeatEmissions

File: app/containers/Products/product-schema.json [hunk 2]
```diff
@@ -24,6 +24,59 @@
         "title": "CIIP (benchmarked) product"
       }
     },
+    "dependencies": {
+      "requiresEmissionAllocation": {
+        "oneOf": [
+          {
+            "properties": {
+              "requiresEmissionAllocation": {
+                "const": false
```
Comment (matthieu-foucault) [inline app/containers/Products/product-schema.json]:
```suggestion
                "const": true
```
It's the opposite: if the product requires emission allocation, then there is no need to specify which electricity/heat emissions are added/substracted as the reported will do the calculation themselves.

File: schema/deploy/tables/product.sql [hunk 3]
```diff
@@ -12,6 +12,10 @@ create table ggircs_portal.product (
   parent integer ARRAY,
   requires_emission_allocation boolean not null default false,
   is_ciip_product boolean not null default true,
+  includes_imported_electricity boolean,
+  includes_exported_electricity boolean,
+  includes_imported_heat boolean,
+  includes_exported_heat boolean,
```
Comment (matthieu-foucault) [inline schema/deploy/tables/product.sql]: See comment re. renaming the properties

File: schema/deploy/tables/product.sql [hunk 4]
```diff
@@ -71,6 +75,10 @@ comment on column ggircs_portal.product.description is 'The description of the p
 comment on column ggircs_portal.product.units is 'The unit of measure for the product';
 comment on column ggircs_portal.product.requires_emission_allocation is 'Boolean value indicates if the product requires allocation of emissions';
 comment on column ggircs_portal.product.is_ciip_product is 'Boolean value indicates if the product is benchmarked and has an associated incentive';
+comment on column ggircs_portal.product.includes_imported_electricity is 'Boolean value indicates if the product should attribute imported electricity to emissions';
+comment on column ggircs_portal.product.includes_exported_electricity is 'Boolean value indicates if the product should attribute exported electricity to emissions';
+comment on column ggircs_portal.product.includes_imported_heat is 'Boolean value indicates if the product should attribute imported heat to emissions';
+comment on column ggircs_portal.product.includes_exported_heat is 'Boolean value indicates if the product should attribute exported heat to emissions';
```
Comment (matthieu-foucault) [inline schema/deploy/tables/product.sql]: Those can be more explicit: `add_imported_electricity_emissions` could be 'Boolean value indicates if imported electricity emissions should be added to the facility emissions when calculating the product emission intensity (applies only to products where requires_emission_allocation is false)'

**Comment by dleard** [comment]
> Sorry for the "let's rename everything" change request

allthings

File: app/containers/Products/ProductCreatorContainer.tsx [hunk 5]
```diff
@@ -27,11 +27,15 @@ export const ProductCreator: React.FunctionComponent<Props> = ({
           units: e.formData.units,
           state: 'active',
           parent: [null],
-          requiresEmissionAllocation: e.formData.requiresEmissionAllocation
+          requiresEmissionAllocation: e.formData.requiresEmissionAllocation,
+          addImportedElectricityEmissions:
+            e.formData.addImportedElectricityEmissions,
+          addExportedElectricity: e.formData.addExportedElectricityEmissions,
```
Comment (matthieu-foucault) [inline app/containers/Products/ProductCreatorContainer.tsx]:
```suggestion
          substractExportedElectricity: e.formData.substractExportedElectricityEmissions,
```

File: app/containers/Products/ProductCreatorContainer.tsx [hunk 6]
```diff
@@ -27,11 +27,15 @@ export const ProductCreator: React.FunctionComponent<Props> = ({
...
+          addImportedHeatEmissions: e.formData.addImportedHeatEmissions,
+          addExportedHeatEmissions: e.formData.addExportedHeatEmissions
```
Comment (matthieu-foucault) [inline app/containers/Products/ProductCreatorContainer.tsx]:
```suggestion
          substractExportedHeatEmissions: e.formData.substractExportedHeatEmissions
```

**Pull request merged**